### DISCUSSION TITLE
Archives don't contain submodules

### DISCUSSION THREAD

**Discussion by SoapSpangledGames** [uNetworking/uWebSockets#1445] [Q&A] [answered]
I downloaded the latest zip file, and typed make at the command line. This is what I got:

$ make
make -C uSockets; \
for FILE in Broadcast HelloWorld ServerName EchoServer BroadcastingEchoServer UpgradeSync UpgradeAsync; do g++ -flto -O3 -lpthread -Wpedantic -Wall -Wextra -Wsign-conversion -Wconversion -std=c++2a -Isrc -IuSockets/src examples/$FILE.cpp -o $FILE uSockets/*.o -lz & done; \
for FILE in HelloWorldThreaded EchoServerThreaded; do g++ -pthread -flto -O3 -lpthread -Wpedantic -Wall -Wextra -Wsign-conversion -Wconversion -std=c++2a -Isrc -IuSockets/src examples/$FILE.cpp -o $FILE uSockets/*.o -lz & done; \
wait
make[1]: Entering directory uWebSockets-master/uSockets'
make[1]: *** No targets specified and no makefile found.  Stop.
make[1]: Leaving directory 'uWebSockets-master/uSockets'
g++: error: uSockets/*.o: No such file or directory
g++: error: uSockets/*.o: No such file or directory
g++: error: uSockets/*.o: No such file or directory
g++: error: uSockets/*.o: No such file or directory
g++: error: uSockets/*.o: No such file or directory
g++: error: uSockets/*.o: No such file or directory
g++: error: uSockets/*.o: No such file or directory
g++: error: uSockets/*.o: No such file or directory
g++: error: uSockets/*.o: No such file or directory
$ 

Any suggestions on what's happening, and how to fix it?

**Comment by ghost** [comment]
GitHub don't pack submodules in generated archives

**Comment by e3dio** [comment, marked as answer]
`git clone https://github.com/uNetworking/uWebSockets --recursive`

**Reply by SoapSpangledGames** [reply]
Thanks. That solved the problem. I also didn't have zlib1g-dev installed on Kubuntu 20.04, which caused compile errors. Installing it made the compile work.
